src/ophyd_async/testing/_one_of_everything.py

```diff
@@ -53,29 +59,58 @@ def float_array_signal(
         ],
         dtype=dtype,
     )
-    return soft_signal_rw(Array1D[dtype], value)
+    return soft_signal_rw(Array1D[dtype], value, name)
+
+
+class DummyController(DetectorController):
+    def get_deadtime(self, exposure: float | None) -> float: ...
+
+    async def prepare(self, trigger_info): ...
+
+    async def arm(self) -> None: ...
+
+    async def wait_for_idle(self): ...
+
+    async def disarm(self): ...
+
 
+class DummyWriter(DetectorWriter):
+    async def open(self, multiplier: int = 1):
+        return {}
 
-class OneOfEverythingDevice(Device):
+    def observe_indices_written(self, timeout: float = 10.0):
+        raise NotImplementedError()
+
+    async def get_indices_written(self) -> int:
+        return 0
+
+    def collect_stream_docs(self, indices_written: int):
+        raise NotImplementedError()
+
+    async def close(self) -> None: ...
+
+
+class OneOfEverythingDevice(StandardDetector):
```

Comment:
Rather than making it a `StandardDetector`, would it be better to make it a `StandardReadable`? Then you wouldn't need the DummyController and DummyWriter...